docs/src/main/sphinx/functions/table.rst

```diff
@@ -47,6 +47,58 @@ by the specified catalog, the query fails.
 The table function name is resolved case-insensitive, analogically to scalar
 function and table resolution in Trino.
 
+Arguments
+^^^^^^^^^
+
+There are three types of arguments.
+
+1. Scalar arguments
+
+They must be constant expressions, and they can be of any SQL type, which is
+compatible with the declared argument type::
+
+    factor => 42
+
+2. Descriptor arguments
+
+Descriptors consist of fields with names and optional data types::
+
+    schema => DESCRIPTOR(id BIGINT, name VARCHAR)
+    columns => DESCRIPTOR(date, status, comment)
+
+To pass ``null`` for descriptor, use::
+
+    schema => CAST(null AS DESCRIPTOR)
+
+3. Table arguments
+
+You can pass a table name, or a query. Use the keyword ``TABLE``::
+
+    input => TABLE(orders)
+    data => TABLE(SELECT * FROM region, nation WHERE region.regionkey = nation.regionkey)
+
+If the table argument is declared as :ref:`set semantics<set_or_row_semantics>`,
+you can specify partitioning and ordering. Each partition is processed
+independently by the table function. If you do not specify partitioning, the
+argument is processed as a single partition.
+
+Additionally, if the table argument is declared as
```

Comment:
do we really want to link again and have a new paragraph.. its about the same "set semantics" in both paragraphs

Reply:
merged